### PR TITLE
Add CI/CD and transition from Leiningen to Deps

### PR DESCRIPTION
**Pull request by mortenschioler**
- Use deps.edn instead of Leiningen
- Add automatic tests on pull requests
- Add automatic deploy on tag release
- Update README

@aengelberg

### CODE REVIEW

File: .github/workflows/publish_on_release.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            ~/.gitlibs
+            ~/.deps.clj
+          # List all files containing dependencies:
+          key: cljdeps-${{ hashFiles('deps.edn') }}
+          restore-keys: cljdeps-
+      # SETUP END
+
+      - name: Build pipeline
+        run: >-
+          clojure -X:build pipeline :lib io.github.mortenschioler/foo :version '"${{github.ref_name}}"'
```
Comment (aengelberg) [inline .github/workflows/publish_on_release.yml]: did you intend for this to be `foo`?

**Review by aengelberg — APPROVED** [review]

File: .github/workflows/publish_on_release.yml (hunk 1 above)
Comment (mortenschioler) [reply .github/workflows/publish_on_release.yml]: No, good eye! I realized that I forgot to change this on my bike trip home, and thought I'd change it tomorrow, but you were too quick for that! 😁

Comment (mortenschioler) [reply .github/workflows/publish_on_release.yml]: Fixed in 5b013a9.

**Comment by mortenschioler** [comment]
Thanks for the review Alex. Everything worked like a charm, from build-per-pull-request to clojars deploy on release publish to README badges. Time to start working on the content!

**Pull request merged**